lib/providers/products_provider.dart

```diff
@@ -48,9 +51,29 @@ class ProductsProvider with ChangeNotifier {
   Product findById(String id) => _items.firstWhere((prod) => prod.id == id);
 
   /// addProduct is called when the user adds a new product in the shop
-  void addProduct(Product product) {
-    _items.add(product);
-    notifyListeners();
+  Future<void> addProduct(Product product) {
+    const url = 'https://shop-app-4ed38-default-rtdb.firebaseio.com/products.json';
+    return http.post(
+      url,
+      body: json.encode({
+        'title': product.title,
```

Comment:
this `product.title` seems repeated in `.then` block, same with `product.description` and others. extract them as variables instead.

Reply:
It is different. In `json.encode` it is a map while in `.then` block, it is the `Product` class.